SynthSeg/training_with_tfrecords.py

```diff
@@ -50,6 +69,12 @@ def training(opts: TrainingOptions) -> tf.keras.callbacks.History:
         strategy = tf.distribute.MirroredStrategy()
     else:
         raise NotImplementedError(f"The '{opts.strategy}' strategy is not implemented.")
+    
+    
+    if opts.mlflow:
+        from .mlflow_utils import set_mlflow_tracking
+        set_mlflow_tracking()
+        mlflow.start_run(log_system_metrics=True)
```

Comment:
Taking into account the other comments, i would suggest something like this
```suggestion
        import mlflow
        
        def handler(signum, frame):
            print("Received cancellation signal. Cleaning up...", signum)
            # Add cleanup actions here, such as saving model state or logging metrics
            # For example, you can log an intermediate metric to MLflow before exiting
            mlflow.end_run("KILLED")
                
        mlflow.tensorflow.autolog(disable=True)        
        mlflow.start_run(log_system_metrics=True)
        signal.signal(signal.SIGINT, handler)
        signal.signal(signal.SIGTERM, handler)  
```

Reply:
Nice suggestion! Implemented and tested it, works fine:)